pkg/controller/scheduler.go

```diff
@@ -868,6 +868,10 @@ func (c *Controller) rollback(canary *flaggerv1.Canary, canaryController canary.
 
 	c.recorder.SetWeight(canary, primaryWeight, canaryWeight)
 
+	if ok := c.runConfirmFinalizingHook(canary, flaggerv1.CanaryPhaseFailed); !ok {
```

Comment:
`rollback()` is meant to roll a Canary analysis back, why would it call the the webhooks mean to confirm canary finalization, which is supposed to indicate that the canary has been promoted and thus being finalized?

Reply:
updated in the description.